themes/classic/modules/ps_newproducts/views/templates/hook/ps_newproducts.tpl

```diff
@@ -22,7 +22,7 @@
  * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
  * International Registered Trademark & Property of PrestaShop SA
  *}
-
+{if $products}
```

Comment:
Not sure it's the best solution to simply hide/remove the whole section
It might be your way of displaying it, but some other merchants would prefer a text in place of product list

Both are possible and valid point of views, but it shouldn't be force in the core template It would be more interesting to manage this through a configuration in the module to adapt the behaviour

besides as you said in the description this can be overridden in a module or a theme template, so if you need to change this behaviour in one your shop that would be more fitting for now no?